xcube/core/resampling/affine.py

```
@@ -158,12 +158,18 @@ def resample_dataset(
                 aggregator = np.nanmean
                 # forman: changed default from True to False (v1.6, 2024-06-05)
                 recover_nan = False
+            if hasattr(var.data, "chunksize"):
+                chunksize = list(var.data.chunksize)
+                chunksize[-2] = tile_height
+                chunksize[-1] = tile_width
+            else:
+                chunksize = (tile_height, tile_width)
```

This is a surprising behaviour. Parameters explicitly passed into a function should be considered or the user should be informed by raising or warning. If `tile_size` is `None`, _then_ you can fallback to default values derived from internal data chunking.

---

`tile_size` comes from `target_gm.tile_size` in line 96, and this is never `None`. Here, I have not changes a lot. I only cover the case for a 3D datacube, where aggregation is needed in the reprojection process. Then I use `affine_transform_dataset`, which without these changes would change my chunking along the 3rd axis to 1, even though the chunksize beforehand was larger. The else case is needed for `var.data` being a numpy array.

---

Ok, that's fine. However, then we should pass a flag to indicate that the tile size was not user provided. I just want to avoid the case where `tile_size` supplied by the user is silently ignored.

---

Ok!